build.rs

```diff
@@ -1,56 +1,49 @@
+use std::{env, error::Error, fs, path::PathBuf};
+
+use git_version::git_version;
 use semver::Version;
-use std::{env, error::Error, fs, path::Path, path::PathBuf, process::Command};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    // Put the linker script somewhere the linker can find it
-    let out = &PathBuf::from(env::var("OUT_DIR")?);
-    let mut linker_script = fs::read_to_string("defmt.x.in")?;
-    let hash = Command::new("git")
-        .args(&["rev-parse", "HEAD"])
-        .output()
-        .ok()
-        .and_then(|output| {
-            if output.status.success() {
-                String::from_utf8(output.stdout).ok()
-            } else {
-                None
-            }
-        });
-    let version = if let Some(hash) = hash {
-        hash
-    } else {
-        assert!(
-            !Path::new(".git").exists(),
-            "you need to install the `git` command line tool to use the git version of `defmt`"
-        );
-
-        // no git info -> assume crates.io
-        let semver = Version::parse(&std::env::var("CARGO_PKG_VERSION")?)?;
-        if semver.major == 0 {
+    const GIT_DESCRIBE: &str = git_version!(args = ["--long", "--match", "v*"], cargo_prefix = "");
+    let version = match Version::parse(GIT_DESCRIBE) {
```

Comment:
super nitpick: the match body reads a bit unintuitive if you're not aware that `Version::parse()` parses *semver* strings. I'd keep the full crate qualifier here, e.g.

```suggestion
    let version = match semver::Version::parse(GIT_DESCRIBE) {
```
(and remove the import at the top)

Reply:
👍🏾

Reply:
Addressed in 9d5f5fe